### PR TITLE
8309334: ProcessTools.main() does not properly set thread names when using the virtual thread wrapper

### PR DESCRIPTION
**Pull request by plummercj**
Normally when a virtual thread wrapper is used to run a test, the main thread is renamed to "old-m-a-i-n" and the new virtual thread that will act as the main thread is named "main". Neither is being done by `ProcessTools.main()`. This can cause problems for tests that expect the main thread that the test is running in to be called "main". It is instead left unnamed. This is causing the following 4 tests to fail:

com/sun/jdi/JdbMethodExitTest.java
com/sun/jdi/JdbStepTest.java
com/sun/jdi/JdbStopThreadTest.java
com/sun/jdi/JdbStopThreadidTest.java

These tests also fail due to [JDK-8309397](https://bugs.openjdk.org/browse/JDK-8309397), which will be fixed after this CR, and also com/sun/jdi/JdbMethodExitTest.java fails due to [JDK-8309396](https://bugs.openjdk.org/browse/JDK-8309396), which will also subsequently be fixed.

Note this fix messed up one runtime test. It was expecting an exception message to mention the "main" thread rather than "old-m-a-i-n". Loosening the exception message matching pattern a bit solved the problem.

Testing was done by running all of tier1 and tier5.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309334](https://bugs.openjdk.org/browse/JDK-8309334): ProcessTools.main() does not properly set thread names when using the virtual thread wrapper

### Reviewers
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**) ⚠️ Review applies to [de5ca147](https://git.openjdk.org/jdk/pull/14292/files/de5ca1471a4df9c19290a2c5aa4eb3c16884faaf)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**) ⚠️ Review applies to [de5ca147](https://git.openjdk.org/jdk/pull/14292/files/de5ca1471a4df9c19290a2c5aa4eb3c16884faaf)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14292/head:pull/14292` \
`$ git checkout pull/14292`

Update a local copy of the PR: \
`$ git checkout pull/14292` \
`$ git pull https://git.openjdk.org/jdk.git pull/14292/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14292`

View PR using the GUI difftool: \
`$ git pr show -t 14292`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14292.diff">https://git.openjdk.org/jdk/pull/14292.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14292#issuecomment-1574345059)

### CODE REVIEW

**Review by alexmenkov — APPROVED** [review]

**Review by lmesnik — APPROVED** [review]

File: test/lib/jdk/test/lib/process/ProcessTools.java [hunk 1]
```diff
@@ -894,7 +896,7 @@ public static void main(String[] args) throws Throwable {
             // MainThreadGroup used just as a container for exceptions
             // when main is executed in virtual thread
             MainThreadGroup tg = new MainThreadGroup();
-            Thread vthread = startVirtualThread(() -> {
+            Thread vthread = createVirtualThread(() -> {
```
Comment (AlanBateman) [inline test/lib/jdk/test/lib/process/ProcessTools.java:899]: You can replace this with `Thread.ofVirtual().unstarted(() -> {` and remove the helper method. I think that helper method dates from the jtreg wrapper solution.

Comment (plummercj) [reply test/lib/jdk/test/lib/process/ProcessTools.java:899]: Yes, I already brought this up with Leonid. I was going to file a separate bug for it, but I can fix it with this CR if you'd like.

**Review by sspitsyn — APPROVED** [review]
Looks good.
Thanks,
Serguei

**Review by AlanBateman — APPROVED** [review]

**Comment by plummercj** [comment]
Thanks for the reviews Leonid, Alan, Serguei and Alex!

/integrate

**Pull request closed without merging**